atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepMetrics.java

```diff
@@ -15,62 +15,99 @@
  */
 package com.palantir.atlasdb.sweep;
 
-import com.codahale.metrics.Meter;
+import java.util.Map;
+import java.util.Optional;
+
+import com.codahale.metrics.Gauge;
 import com.codahale.metrics.MetricRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.SweepResults;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.logging.LoggingArgs;
 import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.logsafe.Arg;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
 
 @SuppressWarnings("checkstyle:FinalClass")
 public class SweepMetrics {
-    private final MetricsManager metricsManager = new MetricsManager();
-
-    private final TableSpecificHistogramMetric cellsSweptHistogram =
-            new TableSpecificHistogramMetric("cellTimestampPairsExamined");
-    private final TableSpecificHistogramMetric cellsDeletedHistogram =
-            new TableSpecificHistogramMetric("staleValuesDeleted");
+    private final TaggedMetricRegistry metricRegistry = new MetricsManager().getTaggedRegistry();
 
-    private final MeterMetric cellsSweptMeter = new MeterMetric("cellsSwept");
-    private final MeterMetric cellsDeletedMeter = new MeterMetric("cellsDeleted");
-    private final MeterMetric sweepErrorMeter = new MeterMetric("sweepError");
+    private final SweepMetric cellsSweptMetric = new SweepMetric("cellTimestampPairsExamined");
```

Comment:
I'm not entirely sure how tagged metrics work but I think there's a problem with this approach.  Both the incremental and end of table metrics will have the same name.  We can filter by tag to see the end of table results, but we can't filter out the end of table results from the then incremental (untagged) results.

When we create sweep metrics we should pass in whether it is incremental or full results and the table name and register the appropriate metrics.  This may also let us tag the incremental results with a table name.

We can ask the MetricsRegistry for the Metric object each time if we don't want to store a map of them.

Reply:
Alternatively we can add more tags as I've done now. I feel that's way cleaner, but feel free to push back.